File: ambeth/Ambeth.Cache/ambeth/template/DataObjectTemplate.cs
using De.Osthus.Ambeth.Cache;
using De.Osthus.Ambeth.Ioc.Annotation;
using De.Osthus.Ambeth.Merge;
using De.Osthus.Ambeth.Model;
using System;

namespace De.Osthus.Ambeth.Template
{
    public class DataObjectTemplate
    {
        [Autowired]
        public ICacheModification CacheModification { protected get; set; }

        [Autowired]
        public IRevertChangesHelper RevertChangesHelper { protected get; set; }
        
        public void ToBeUpdatedChanged(IDataObject obj, bool previousValue, bool currentValue)
        {
            if (previousValue && !currentValue && !CacheModification.ActiveOrFlushing)
            {
                RevertChangesHelper.RevertChanges(obj);
            }
        }
    }
}

File: jambeth/jambeth-cache/src/main/java/de/osthus/ambeth/template/DataObjectTemplate.java
package de.osthus.ambeth.template;

import de.osthus.ambeth.cache.ICacheModification;
import de.osthus.ambeth.ioc.annotation.Autowired;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.merge.IRevertChangesHelper;
import de.osthus.ambeth.model.IDataObject;

public class DataObjectTemplate
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Autowired
	protected ICacheModification cacheModification;

	@Autowired
	protected IRevertChangesHelper revertChangesHelper;

	public final void toBeUpdatedChanged(IDataObject obj, boolean previousValue, boolean currentValue)
	{
		if (previousValue && !currentValue && !cacheModification.isActiveOrFlushing())
		{
			revertChangesHelper.revertChanges(obj);
		}
	}
}
